app/components/UI/FiatOnRampAggregator/Views/Region.js

```diff
@@ -18,47 +18,41 @@ import RegionAlert from '../components/RegionAlert';
 const Region = () => {
 	const navigation = useNavigation();
 	const { colors } = useTheme();
-	const { setSelectedCountry, setSelectedRegion, selectedCountry, selectedRegion } = useFiatOnRampSDK();
-	const [isRegionModalVisible, toggleRegionModal, , hideRegionModal] = useModalHandler(false);
+	const { selectedRegion, setSelectedRegion } = useFiatOnRampSDK();
+	const [isRegionModalVisible, , showRegionModal, hideRegionModal] = useModalHandler(false);
 
 	const [showAlert, setShowAlert] = useState(false);
 	const [selectedUnsupportedLocation, setSelectedUnsupportedLocation] = useState({});
 	const [{ data, isFetching, error }] = useSDKMethod('getCountries');
-
 	useEffect(() => {
 		navigation.setOptions(getFiatOnRampAggNavbar(navigation, { title: 'Region', showBack: false }, colors));
 	}, [navigation, colors]);
 
-	const handleRegionButton = () => {
-		toggleRegionModal();
-	};
-
 	const handleOnPress = useCallback(() => {
 		navigation.navigate('PaymentMethod');
 	}, [navigation]);
 
-	const handleCountryPress = (country) => {
-		setSelectedCountry(country);
-		hideRegionModal();
-	};
-
 	const handleRegionPress = useCallback(
-		(region, country) => {
-			if (region.unsupported) {
-				setShowAlert(true);
-				setSelectedUnsupportedLocation(region);
-			} else {
-				setSelectedRegion(region);
-				setSelectedCountry(country);
-				hideRegionModal();
-			}
+		(region) => {
+			setSelectedRegion(region);
+			hideRegionModal();
 		},
-		[hideRegionModal, setSelectedCountry, setSelectedRegion]
+		[hideRegionModal, setSelectedRegion]
 	);
 
-	const handleUnsetRegion = useCallback(() => {
-		setSelectedRegion(undefined);
-	}, [setSelectedRegion]);
+	const updatedRegion = useMemo(() => {
+		if (!selectedRegion || !data) return null;
+		const allRegions = data.reduce((acc, region) => [...acc, region, ...(region.states || [])], []);
```

Comment:
Since you are flattening the list of regions, I think It would make sense to filter out regions with states as those regions can not be selected, the backend is flagging them as "unsupported" so I suggest you can do chain a filter to allRegions as following:
```suggestion
		const allRegions = data.reduce((acc, region) => [...acc, region, ...(region.states || [])], [])
		                                        .filter(({states}) => !states)
```
by doing that you create a flat list of leaf nodes only.

Reply:
We just do this to find the selected region. By filtering out the "states" region we break the behavior of the modal selector.